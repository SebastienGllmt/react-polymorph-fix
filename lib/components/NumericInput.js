'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NumericInput = undefined;

var _flow2 = require('lodash/flow');

var _flow3 = _interopRequireDefault(_flow2);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _createRef = require('create-react-ref/lib/createRef');

var _createRef2 = _interopRequireDefault(_createRef);

var _withTheme = require('./HOC/withTheme');

var _themes = require('../utils/themes');

var _ = require('.');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

// external libraries

// $FlowFixMe


// internal utility functions


// import constants


var NumericInputBase = function (_Component) {
  _inherits(NumericInputBase, _Component);

  // define static properties
  function NumericInputBase(props) {
    _classCallCheck(this, NumericInputBase);

    var _this = _possibleConstructorReturn(this, (NumericInputBase.__proto__ || Object.getPrototypeOf(NumericInputBase)).call(this, props));

    _this.onChange = function (event) {
      event.preventDefault();
      var _this$props = _this.props,
          onChange = _this$props.onChange,
          disabled = _this$props.disabled;

      if (disabled) {
        return;
      }

      // it is crucial to remove whitespace from input value
      // with String.trim()
      var processedValue = _this._processValue(event.target.value.trim(), event.target.selectionStart);

      // if the processed value is the same, then the user probably entered
      // invalid input such as nonnumeric characters, do not call onChange
      if (processedValue === _this.state.oldValue) {
        return;
      }

      if (onChange) {
        onChange(processedValue, event);
      }
    };

    _this.focus = function () {
      var inputElement = _this.inputElement;

      if (!inputElement.current) return;
      inputElement.current.focus();
    };

    _this._setError = function (error) {
      var setError = _this.props.setError;

      // checks for setError func from FormField component
      // if this NumericInput instance is rendered within FormField's render prop,
      // FormField's local state.error will also be set via props.setError

      if (setError) setError(error);
      // also set (this: NumericInput)'s state.error
      _this.setState({ error: error });
    };

    var context = props.context,
        minValue = props.minValue,
        maxBeforeDot = props.maxBeforeDot,
        maxAfterDot = props.maxAfterDot,
        themeId = props.themeId,
        theme = props.theme,
        themeOverrides = props.themeOverrides;


    var minValueIsNum = typeof minValue === 'number';
    // if minValue is a number and user supplied maxBeforeDot and/or maxAfterDot
    if (minValue && minValueIsNum && (maxBeforeDot || maxAfterDot)) {
      // check combination of values for validity
      _this._validateLimitProps(minValue, maxBeforeDot, maxAfterDot);
    }

    // define ref
    _this.inputElement = (0, _createRef2.default)();

    _this.state = {
      composedTheme: (0, _themes.composeTheme)((0, _themes.addThemeId)(theme || context.theme, themeId), (0, _themes.addThemeId)(themeOverrides, themeId), context.ROOT_THEME_API),
      caretPosition: 0,
      separatorsCount: 0,
      error: '',
      oldValue: ''
    };
    return _this;
  }
  // declare ref types


  _createClass(NumericInputBase, [{
    key: 'componentDidMount',
    value: function componentDidMount() {
      var inputElement = this.inputElement;
      // check for autoFocus prop

      if (this.props.autoFocus) this.focus();

      // Set last input caret position on updates
      if (inputElement && inputElement.current) {
        this.setState({ caretPosition: inputElement.current.selectionStart });
      }
    }
  }, {
    key: 'componentWillReceiveProps',
    value: function componentWillReceiveProps(nextProps) {
      (0, _themes.didThemePropsChange)(this.props, nextProps, this.setState.bind(this));
    }
  }, {
    key: 'componentDidUpdate',
    value: function componentDidUpdate(prevProps, prevState) {
      var inputElement = this.inputElement;

      if (inputElement && inputElement.current !== document.activeElement) {
        return;
      }

      // caret position calculation after separators injection
      var caretPosition = void 0;
      // prevent unnecessary changes on re-rendering
      if (this.state.oldValue !== prevState.oldValue || this.state.caretPosition !== prevState.caretPosition) {
        if (this.state.separatorsCount !== prevState.separatorsCount && this.state.separatorsCount - prevState.separatorsCount <= 1 && this.state.separatorsCount - prevState.separatorsCount >= -1) {
          caretPosition = this.state.caretPosition + (this.state.separatorsCount - prevState.separatorsCount);
        } else {
          caretPosition = this.state.caretPosition;
        }
        caretPosition = caretPosition >= 0 ? caretPosition : 0;

        if (inputElement && inputElement.current) {
          inputElement.current.selectionEnd = caretPosition;
          inputElement.current.selectionStart = caretPosition;
        }
      }
    }
  }, {
    key: '_validateLimitProps',
    value: function _validateLimitProps(minValue, maxBeforeDot, maxAfterDot) {
      if (typeof minValue !== 'number') return;
      var maxBeforeDotIsNum = typeof maxBeforeDot === 'number';
      var maxAfterDotIsNum = typeof maxAfterDot === 'number';
      // if minValue is a float, it will split at the decimal
      // trailing zeros are dropped with parseFloat
      var minValParts = parseFloat(minValue).toString().split('.');

      // if minValParts array has length of 2, it is a float
      if (minValParts.length >= 2) {
        var minValBeforeDot = minValParts[0];
        var minValAfterDot = minValParts[1];

        // if the number of integers in minValue is greater than maxBeforeDot
        if (maxBeforeDot && maxBeforeDotIsNum && minValBeforeDot.length > maxBeforeDot) {
          // the combo is incompatible, throw error
          var _error = '\n          minValue: ' + minValue + ' exceeds the limit of maxBeforeDot: ' + maxBeforeDot + '.\n          Adjust the values of these properties.\n        ';
          throw new Error(_error);
          // if the number of decimal spaces in minValue is greater than maxBeforeDot
        } else if (maxAfterDot && maxAfterDotIsNum && minValAfterDot.length > maxAfterDot) {
          var _error2 = '\n          minValue: ' + minValue + ' exceeds the limit of maxAfterDot: ' + maxAfterDot + '.\n          Adjust the values of these properties.\n        ';
          throw new Error(_error2);
        }
      }
    }
  }, {
    key: '_processValue',
    value: function _processValue(value, position) {
      return (0, _flow3.default)([this._enforceNumericValue, this._parseToParts, this._enforceValueLimits, this._separate]).call(this, value, position);
    }
  }, {
    key: '_enforceNumericValue',
    value: function _enforceNumericValue(value, position) {
      var regex = /^[0-9.,]+$/;
      var isValueRegular = regex.test(value);
      var maxBeforeDot = this.props.maxBeforeDot;

      var handledValue = void 0;
      var lastValidValue = this.state.oldValue;
      if (!isValueRegular && value !== '') {
        // input contains invalid value
        // e.g. 1,00AAbasdasd.asdasd123123
        // - reject it and show last valid value
        handledValue = lastValidValue || '0.000000';
        position -= 1;
      } else if (!this._isNumeric(value)) {
        // input contains comma separated number
        // e.g. 1,000,000.123456
        // - make sure commas and caret are at correct position
        var splitedValue = value.split('.');

        // check if input value contains more than one decimal
        if (splitedValue.length === 3) {
          var splitedOldValue = lastValidValue.split('.');
          var _beforeDot = splitedValue[0] + splitedValue[1];
          // variable for value before the decimal containing a comma. Ex: 1,425
          var beforeDotWithoutComma = _beforeDot;

          // if comma exists, remove before comparing length in next if-else statement
          if (_beforeDot.includes(',')) {
            var beforeComma = _beforeDot.slice(0, _beforeDot.indexOf(','));
            var afterComma = _beforeDot.slice(_beforeDot.indexOf(',') + 1);
            beforeDotWithoutComma = beforeComma + afterComma;
          }
          if (!_beforeDot.includes(',') && splitedOldValue[0].length < _beforeDot.length || _beforeDot.includes(',') && splitedOldValue[0].length < beforeDotWithoutComma.length) {
            // dot is in decimal part
            position -= 1;
            handledValue = _beforeDot + '.' + splitedValue[2];
            _beforeDot = _beforeDot.replace(/,/g, '');
            // prevent replace dot if length before dot is greater then max before dot length
            if (maxBeforeDot && _beforeDot.length > maxBeforeDot) {
              handledValue = lastValidValue;
            }
          } else {
            handledValue = splitedValue[0] + '.' + splitedValue[1] + splitedValue[2];
            // Second dot was entered after current one -> stay in same position (swallow dot)
            if (position > _beforeDot.length + 1) {
              position -= 1;
            }
          }
        } else if (splitedValue.length === 2 && splitedValue[0] === '' && splitedValue[1] === '') {
          // special case when dot is inserted in an empty input
          // - return 0.|00000
          handledValue = '0.000000';
          position = 2; // position caret after the dot
        } else if (value !== '') {
          // special case when user selects part of an input value and hits ',' key
          // - reject it and show last valid value
          handledValue = lastValidValue;
        }
      }

      var lastInsertedCharacter = value.substring(position - 1, position);
      if (lastInsertedCharacter === ',') {
        // prevent move caret position on hit ','
        position -= 1;
      }

      return !this._isNumeric(value) ? { value: handledValue, position: position } : { value: value, position: position };
    }
  }, {
    key: '_parseToParts',
    value: function _parseToParts(data) {
      var value = data.value;
      var position = data.position;

      // show placeholder on select all and delete/backspace key action
      if (!value) return;

      var beforeDot = void 0;
      var afterDot = void 0;

      if (data.value.length > 1 && value.split('.').length < 2) {
        // handle numbers deletion from both integer and decimal parts at once
        beforeDot = value.substring(0, position) || '0';
        afterDot = value.substring(position, value.length);
      } else {
        // split float number to integer and decimal part - regular way
        var splitedValue = value.split('.');
        beforeDot = splitedValue[0] ? splitedValue[0] : '0';
        afterDot = splitedValue[1] ? splitedValue[1] : '000000';
      }

      // remove leading zero and update caret position
      if (value.charAt(0) === '0' && parseInt(beforeDot.replace(/,/g, ''), 10) > 0) {
        beforeDot = parseInt(beforeDot.replace(/,/g, ''), 10);
        if (position !== 2) {
          position = 0;
        } else {
          position = 1;
        }
      } else if (parseInt(beforeDot.replace(/,/g, ''), 10) === 0) {
        beforeDot = parseInt(beforeDot.replace(/,/g, ''), 10);
      }

      return { value: value, position: position, parts: { beforeDot: beforeDot, afterDot: afterDot } };
    }

    // enforces props.maxValue and props.minValue

  }, {
    key: '_enforceValueLimits',
    value: function _enforceValueLimits(data) {
      if (!data) return;

      var _props = this.props,
          minValue = _props.minValue,
          maxValue = _props.maxValue,
          enforceMax = _props.enforceMax,
          enforceMin = _props.enforceMin,
          maxAfterDot = _props.maxAfterDot;
      var position = data.position;

      // enforce props.maxBeforeDot and props.maxAfterDot

      var valueWithDecimalRestrictions = this._enforceDecimalRestrictions(data);

      // creates floating point number equal to valueWithDecimalRestrictions (string)
      // will be used for value comparisons against props.maxValue and props.minValue if applicable
      var valueWithoutSeparators = parseFloat(valueWithDecimalRestrictions.replace(/,/g, ''));

      // if input value is greater than props.maxValue, throw error
      if (maxValue && valueWithoutSeparators > maxValue) {
        var formattedMaxVal = maxValue.toFixed(maxAfterDot || 6).toString();
        this._setError('Maximum amount is ' + formattedMaxVal);

        // if user passes enforceMax=true, restrict input value to props.maxValue
        if (enforceMax) {
          this.setState({ caretPosition: position });
          return formattedMaxVal;
        }
        // if input value is below props.minValue, throw error
      } else if (minValue && valueWithoutSeparators < minValue) {
        var formattedMinVal = minValue.toFixed(maxAfterDot || 6).toString();
        this._setError('Minimum amount is ' + formattedMinVal);

        // if props.enforceMin=true, restrict input value to props.minValue
        if (enforceMin) {
          this.setState({ caretPosition: position });
          return formattedMinVal;
        }
        // if input value has no errors, clear state.error
      } else if (this.state.error !== '') {
        this._setError('');
      }

      // update caret in state
      this.setState({ caretPosition: position });

      // input value w/ decimal restrictions is passed along
      // to this._separate without value restrictions
      return valueWithDecimalRestrictions;
    }

    // enforces props.maxBeforeDot and props.maxAfterDot

  }, {
    key: '_enforceDecimalRestrictions',
    value: function _enforceDecimalRestrictions(data) {
      var _props2 = this.props,
          maxBeforeDot = _props2.maxBeforeDot,
          maxAfterDot = _props2.maxAfterDot;
      var beforeDot = data.parts.beforeDot;
      var afterDot = data.parts.afterDot;

      // preventing numbers with more than maxBeforeDot units
      // - return first maxBeforeDot numbers (with comma separators)

      if (maxBeforeDot && beforeDot) {
        // max number of commas depending on max number of characters before dot
        var numberOfCommas = maxBeforeDot % 3 > 0 ? parseInt(maxBeforeDot / 3, 10) : parseInt(maxBeforeDot / 3, 10) - 1;
        var maxBeforeDotWithSeparator = maxBeforeDot + numberOfCommas;
        if (beforeDot.length > maxBeforeDotWithSeparator) {
          beforeDot = beforeDot.substring(0, maxBeforeDotWithSeparator);
        }
      }

      // remove commas from decimal part
      // (e.g. 123,23,2.002000 -> dot after 2.character reproduce 12.3,23,2)
      afterDot = afterDot.replace(/,/g, '');
      // preventing numbers with more than maxAfterDot units - return first maxAfterDot numbers
      if (maxAfterDot && afterDot && afterDot.length > maxAfterDot) {
        afterDot = afterDot.substring(0, maxAfterDot);
      }

      // if decimal number has less than maxAfterDot numbers add trailing zeros
      var afterDotLength = afterDot ? afterDot.length : 0;
      if (maxAfterDot && afterDotLength < maxAfterDot) {
        for (afterDotLength; afterDotLength < maxAfterDot; afterDotLength++) {
          afterDot += '0';
        }
      }
      // if maxAfterDot is 0, drop decimal & numbers after decimal, return int
      if (maxAfterDot === 0) {
        return beforeDot;
      }

      // return input value w/decimal restrictions as a string
      return beforeDot + '.' + afterDot;
    }
  }, {
    key: '_separate',
    value: function _separate(value) {
      this.setState({ oldValue: value });
      // value will not contain '.' if maxAfterDot is 0, return early
      if (value && !value.includes('.')) {
        return value;
      }
      if (!value) {
        this.setState({ separatorsCount: 0 });
      }
      if (value) {
        var splitedValue = value.split('.');
        var separatedValue = splitedValue[0].replace(/,/g, '').split('').reverse().join('').replace(/(\d{3}\B)/g, '$1,').split('').reverse().join('');
        var newSeparatorsCount = (separatedValue.match(/,/g) || []).length;
        this.setState({ separatorsCount: newSeparatorsCount });
        return separatedValue + '.' + splitedValue[1];
      }
    }
  }, {
    key: '_isNumeric',
    value: function _isNumeric(value) {
      var replacedValue = value.replace(/,/g, '');
      // eslint-disable-next-line no-restricted-globals
      return !isNaN(parseFloat(replacedValue)) && isFinite(replacedValue);
    }
  }, {
    key: 'render',
    value: function render() {
      // destructuring props ensures only the "...rest" get passed down
      var _props3 = this.props,
          skin = _props3.skin,
          theme = _props3.theme,
          themeOverrides = _props3.themeOverrides,
          onChange = _props3.onChange,
          error = _props3.error,
          context = _props3.context,
          maxValue = _props3.maxValue,
          minValue = _props3.minValue,
          maxBeforeDot = _props3.maxBeforeDot,
          maxAfterDot = _props3.maxAfterDot,
          rest = _objectWithoutProperties(_props3, ['skin', 'theme', 'themeOverrides', 'onChange', 'error', 'context', 'maxValue', 'minValue', 'maxBeforeDot', 'maxAfterDot']);

      var InputSkin = skin || context.skins[_.IDENTIFIERS.INPUT];

      return _react2.default.createElement(InputSkin, _extends({
        error: error || this.state.error,
        inputRef: this.inputElement,
        onChange: this.onChange,
        theme: this.state.composedTheme
      }, rest));
    }
  }]);

  return NumericInputBase;
}(_react.Component);

NumericInputBase.displayName = 'NumericInput';
NumericInputBase.defaultProps = {
  context: (0, _withTheme.createEmptyContext)(),
  enforceMax: false,
  enforceMin: false,
  readOnly: false,
  theme: null,
  themeId: _.IDENTIFIERS.INPUT,
  themeOverrides: {},
  value: ''
};
var NumericInput = exports.NumericInput = (0, _withTheme.withTheme)(NumericInputBase);
//# sourceMappingURL=NumericInput.js.map