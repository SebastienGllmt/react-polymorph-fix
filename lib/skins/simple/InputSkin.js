'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputSkin = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

// external libraries


// components


// skins


// internal utility functions


var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _FormField = require('../../components/FormField');

var _FormFieldSkin = require('./FormFieldSkin');

var _props = require('../../utils/props');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var InputSkin = exports.InputSkin = function InputSkin(props) {
  return _react2.default.createElement(_FormField.FormField, {
    className: props.className,
    disabled: props.disabled,
    label: props.label,
    error: props.error,
    inputRef: props.inputRef,
    skin: _FormFieldSkin.FormFieldSkin,
    theme: props.theme,
    render: function render() {
      return _react2.default.createElement('input', _extends({
        ref: props.inputRef
      }, (0, _props.pickDOMProps)(props), {
        className: (0, _classnames2.default)([props.theme[props.themeId].input, props.disabled ? props.theme[props.themeId].disabled : null, props.error ? props.theme[props.themeId].errored : null]),
        readOnly: props.readOnly
      }));
    }
  });
};
//# sourceMappingURL=InputSkin.js.map