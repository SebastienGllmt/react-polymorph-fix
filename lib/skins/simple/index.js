'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleSkins = undefined;

var _SimpleSkins;

var _components = require('../../components');

var _AutocompleteSkin = require('./AutocompleteSkin');

var _BubbleSkin = require('./BubbleSkin');

var _ButtonSkin = require('./ButtonSkin');

var _CheckboxSkin = require('./CheckboxSkin');

var _FormFieldSkin = require('./FormFieldSkin');

var _HeaderSkin = require('./HeaderSkin');

var _InfiniteScrollSkin = require('./InfiniteScrollSkin');

var _InputSkin = require('./InputSkin');

var _LoadingSpinnerSkin = require('./LoadingSpinnerSkin');

var _ModalSkin = require('./ModalSkin');

var _OptionsSkin = require('./OptionsSkin');

var _ProgressBarSkin = require('./ProgressBarSkin');

var _RadioSkin = require('./RadioSkin');

var _SelectSkin = require('./SelectSkin');

var _SwitchSkin = require('./SwitchSkin');

var _TextAreaSkin = require('./TextAreaSkin');

var _TogglerSkin = require('./TogglerSkin');

var _TooltipSkin = require('./TooltipSkin');

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SimpleSkins = exports.SimpleSkins = (_SimpleSkins = {}, _defineProperty(_SimpleSkins, _components.IDENTIFIERS.AUTOCOMPLETE, _AutocompleteSkin.AutocompleteSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUBBLE, _BubbleSkin.BubbleSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.BUTTON, _ButtonSkin.ButtonSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.CHECKBOX, _CheckboxSkin.CheckboxSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.FORM_FIELD, _FormFieldSkin.FormFieldSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.HEADER, _HeaderSkin.HeaderSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INFINITE_SCROLL, _InfiniteScrollSkin.InfiniteScrollSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.INPUT, _InputSkin.InputSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.LOADING_SPINNER, _LoadingSpinnerSkin.LoadingSpinnerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.MODAL, _ModalSkin.ModalSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.OPTIONS, _OptionsSkin.OptionsSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.PROGRESS_BAR, _ProgressBarSkin.ProgressBarSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.RADIO, _RadioSkin.RadioSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SELECT, _SelectSkin.SelectSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.SWITCH, _SwitchSkin.SwitchSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TEXT_AREA, _TextAreaSkin.TextAreaSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOGGLER, _TogglerSkin.TogglerSkin), _defineProperty(_SimpleSkins, _components.IDENTIFIERS.TOOLTIP, _TooltipSkin.TooltipSkin), _SimpleSkins);
//# sourceMappingURL=index.js.map