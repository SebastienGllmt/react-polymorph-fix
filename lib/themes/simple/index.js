'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SimpleTheme = undefined;

var _SimpleTheme;

var _components = require('../../components');

var _SimpleAutocomplete = require('./SimpleAutocomplete.scss');

var _SimpleAutocomplete2 = _interopRequireDefault(_SimpleAutocomplete);

var _SimpleBubble = require('./SimpleBubble.scss');

var _SimpleBubble2 = _interopRequireDefault(_SimpleBubble);

var _SimpleButton = require('./SimpleButton.scss');

var _SimpleButton2 = _interopRequireDefault(_SimpleButton);

var _SimpleCheckbox = require('./SimpleCheckbox.scss');

var _SimpleCheckbox2 = _interopRequireDefault(_SimpleCheckbox);

var _SimpleFlex = require('./SimpleFlex.scss');

var _SimpleFlex2 = _interopRequireDefault(_SimpleFlex);

var _SimpleFormField = require('./SimpleFormField.scss');

var _SimpleFormField2 = _interopRequireDefault(_SimpleFormField);

var _SimpleGrid = require('./SimpleGrid.scss');

var _SimpleGrid2 = _interopRequireDefault(_SimpleGrid);

var _SimpleGutter = require('./SimpleGutter.scss');

var _SimpleGutter2 = _interopRequireDefault(_SimpleGutter);

var _SimpleHeader = require('./SimpleHeader.scss');

var _SimpleHeader2 = _interopRequireDefault(_SimpleHeader);

var _SimpleInfiniteScroll = require('./SimpleInfiniteScroll.scss');

var _SimpleInfiniteScroll2 = _interopRequireDefault(_SimpleInfiniteScroll);

var _SimpleInput = require('./SimpleInput.scss');

var _SimpleInput2 = _interopRequireDefault(_SimpleInput);

var _SimpleLoadingSpinner = require('./SimpleLoadingSpinner.scss');

var _SimpleLoadingSpinner2 = _interopRequireDefault(_SimpleLoadingSpinner);

var _SimpleModal = require('./SimpleModal.scss');

var _SimpleModal2 = _interopRequireDefault(_SimpleModal);

var _SimpleOptions = require('./SimpleOptions.scss');

var _SimpleOptions2 = _interopRequireDefault(_SimpleOptions);

var _SimpleProgressBar = require('./SimpleProgressBar.scss');

var _SimpleProgressBar2 = _interopRequireDefault(_SimpleProgressBar);

var _SimpleRadio = require('./SimpleRadio.scss');

var _SimpleRadio2 = _interopRequireDefault(_SimpleRadio);

var _SimpleSelect = require('./SimpleSelect.scss');

var _SimpleSelect2 = _interopRequireDefault(_SimpleSelect);

var _SimpleSwitch = require('./SimpleSwitch.scss');

var _SimpleSwitch2 = _interopRequireDefault(_SimpleSwitch);

var _SimpleTextArea = require('./SimpleTextArea.scss');

var _SimpleTextArea2 = _interopRequireDefault(_SimpleTextArea);

var _SimpleToggler = require('./SimpleToggler.scss');

var _SimpleToggler2 = _interopRequireDefault(_SimpleToggler);

var _SimpleTooltip = require('./SimpleTooltip.scss');

var _SimpleTooltip2 = _interopRequireDefault(_SimpleTooltip);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
// import theme IDENTIFIERS constants


// css modules plugin converts all imports below into plain objects


// SimpleTheme is a plain object serving as the default export.
// Each key is named after a component and each key's value
// is the component's corresponding theme. The user can
// pass this entire obj directly to ThemeProvider via the "theme" prop
var SimpleTheme = exports.SimpleTheme = (_SimpleTheme = {}, _defineProperty(_SimpleTheme, _components.IDENTIFIERS.AUTOCOMPLETE, _SimpleAutocomplete2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUBBLE, _SimpleBubble2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.BUTTON, _SimpleButton2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.CHECKBOX, _SimpleCheckbox2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FLEX, _SimpleFlex2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.FORM_FIELD, _SimpleFormField2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GRID, _SimpleGrid2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.GUTTER, _SimpleGutter2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.HEADER, _SimpleHeader2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INFINITE_SCROLL, _SimpleInfiniteScroll2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.INPUT, _SimpleInput2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.LOADING_SPINNER, _SimpleLoadingSpinner2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.MODAL, _SimpleModal2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.OPTIONS, _SimpleOptions2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.PROGRESS_BAR, _SimpleProgressBar2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.RADIO, _SimpleRadio2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SELECT, _SimpleSelect2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.SWITCH, _SimpleSwitch2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TEXT_AREA, _SimpleTextArea2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOGGLER, _SimpleToggler2.default), _defineProperty(_SimpleTheme, _components.IDENTIFIERS.TOOLTIP, _SimpleTooltip2.default), _SimpleTheme);
//# sourceMappingURL=index.js.map