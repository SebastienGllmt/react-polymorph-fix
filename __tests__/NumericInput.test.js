import React from 'react';

import { NumericInput } from '../source/components/NumericInput';
import { InputSkin } from '../source/skins/simple/InputSkin';
import { renderInSimpleTheme } from './helpers/theming';

test('NumericInput renders correctly', () => {
  const component = renderInSimpleTheme(
    <NumericInput
      skin={InputSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NumericInput renders with placeholder', () => {
  const component = renderInSimpleTheme(
    <NumericInput
      placeholder="0.0000"
      skin={InputSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NumericInput is disabled', () => {
  const component = renderInSimpleTheme(
    <NumericInput
      disabled
      skin={InputSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NumericInput is readOnly', () => {
  const component = renderInSimpleTheme(
    <NumericInput
      readOnly
      skin={InputSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NumericInput renders with an error', () => {
  const component = renderInSimpleTheme(
    <NumericInput
      error="Invalid Amount"
      skin={InputSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('NumericInput renders with a value', () => {
  const component = renderInSimpleTheme(
    <NumericInput
      value="555.333"
      skin={InputSkin}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
