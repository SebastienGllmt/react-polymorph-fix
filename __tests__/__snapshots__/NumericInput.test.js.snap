// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericInput is disabled 1`] = `
<div
  className="root disabled"
>
  
  <div
    className="inputWrapper"
  >
    <input
      className="input disabled"
      disabled={true}
      onChange={[Function]}
      readOnly={false}
      value=""
    />
  </div>
</div>
`;

exports[`NumericInput is readOnly 1`] = `
<div
  className="root"
>
  
  <div
    className="inputWrapper"
  >
    <input
      className="input"
      onChange={[Function]}
      readOnly={true}
      value=""
    />
  </div>
</div>
`;

exports[`NumericInput renders correctly 1`] = `
<div
  className="root"
>
  
  <div
    className="inputWrapper"
  >
    <input
      className="input"
      onChange={[Function]}
      readOnly={false}
      value=""
    />
  </div>
</div>
`;

exports[`NumericInput renders with a value 1`] = `
<div
  className="root"
>
  
  <div
    className="inputWrapper"
  >
    <input
      className="input"
      onChange={[Function]}
      readOnly={false}
      value="555.333"
    />
  </div>
</div>
`;

exports[`NumericInput renders with an error 1`] = `
<div
  className="root errored"
>
  <div
    className="error"
  >
    Invalid Amount
  </div>
  <div
    className="inputWrapper"
  >
    <input
      className="input errored"
      onChange={[Function]}
      readOnly={false}
      value=""
    />
  </div>
</div>
`;

exports[`NumericInput renders with placeholder 1`] = `
<div
  className="root"
>
  
  <div
    className="inputWrapper"
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="0.0000"
      readOnly={false}
      value=""
    />
  </div>
</div>
`;
